services/api/src/api/routes/rows.py

```diff
@@ -147,12 +155,14 @@ def __post_init__(
                 if not sources:
                     raise ParquetResponseEmptyError("No parquet files found.")
             with StepProfiler(method="rows.index", step="get the Hub's dataset filesystem"):
-                fs = get_parquet_fs(dataset=self.dataset, hf_token=hf_token)
+                fs = get_hf_fs(hf_token=hf_token)
+            with StepProfiler(method="rows.index", step="get the source URIs"):
```

Comment:
Note I added a profiler step here.

Reply:
(even if it should always be ultra fast, since it's a loop that creates strings). But it's better to be coherent with the rest of the code.